peer-svc/src/store/sql/access/peer.rs

```diff
@@ -0,0 +1,152 @@
+            .bind(hash.as_u32s()[0])
+            .bind(hash.as_u32s()[1])
+            .bind(hash.as_u32s()[2])
+            .bind(hash.as_u32s()[3])
+            .bind(hash.as_u32s()[4])
+            .bind(hash.as_u32s()[5])
+            .bind(hash.as_u32s()[6])
+            .bind(hash.as_u32s()[7])
+            .execute(&mut **conn.inner())
+            .await;
+        match resp {
+            std::result::Result::Ok(_rows) => Ok(true),
+            Err(sqlx::Error::Database(err)) => {
+                if err.is_unique_violation() {
+                    Ok(false)
+                } else {
+                    Err(Error::from(sqlx::Error::Database(err)))
+                }
+            }
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    /// Insert a single interest into the database.
+    pub async fn insert(pool: &SqlitePool, key: &PeerKey) -> Result<bool> {
+        let mut tx = pool.begin_tx().await.map_err(Error::from)?;
+        let new_key = PeerDB::insert_tx(&mut tx, key).await?;
+        tx.commit().await.map_err(Error::from)?;
+        Ok(new_key)
+    }
+
+    /// Insert a multiple interests into the database.
+    pub async fn insert_many(
+        pool: &SqlitePool,
+        items: &[&PeerKey],
+    ) -> Result<InsertResult<PeerKey>> {
+        match items.len() {
+            0 => Ok(InsertResult::new(0)),
+            _ => {
+                let mut results = 0;
+                let mut tx = pool.begin_tx().await.map_err(Error::from)?;
+
+                for item in items.iter() {
+                    PeerDB::insert_tx(&mut tx, item)
+                        .await?
+                        .then(|| results += 1);
+                }
+                tx.commit().await.map_err(Error::from)?;
+                Ok(InsertResult::new(results))
+            }
+        }
+    }
+
+    /// Calculate the Sha256a hash of all keys in the range between left_fencepost and right_fencepost.
+    pub async fn hash_range(
+        pool: &SqlitePool,
+        range: Range<&PeerKey>,
+    ) -> Result<HashCount<Sha256a>> {
+        if range.start >= range.end {
+            return Ok(HashCount::new(Sha256a::identity(), 0));
+        }
+
+        let res: ReconHash = sqlx::query_as(ReconQuery::hash_range(SqlBackend::Sqlite))
+            .bind(range.start.as_bytes())
+            .bind(range.end.as_bytes())
+            .fetch_one(pool.reader())
+            .await
+            .map_err(Error::from)?;
+        let bytes = res.hash();
+        Ok(HashCount::new(Sha256a::from(bytes), res.count()))
+    }
+
+    /// Find the keys in the range
+    pub async fn range(
+        pool: &SqlitePool,
+        range: Range<&PeerKey>,
+        offset: usize,
+        limit: usize,
+    ) -> Result<Vec<PeerKey>> {
+        let query = sqlx::query(ReconQuery::range());
+        let rows = query
+            .bind(range.start.as_bytes())
+            .bind(range.end.as_bytes())
+            .bind(limit as i64)
+            .bind(offset as i64)
```

Comment:
Hmm, good catch, however I don't think we can just use i64::MAX safely either. If we change the limit or offset values the caller passes in they will not be able to page through values. It appears sqlite does not support u64 so I think we need to use u32 for all limit offset pagination indexes we use. This seems sane as a limit or offset larger than u32 is quite large. I'll do this is a seperate PR.

Reply:
Well u32 won't work its too small to be able to count the data. So either we need to use i64 (aka u63) or use Option for limit offset and handle it explicitly in queries. I think we should use i64. Thoughts?

Reply:
hmm. yeah, I think `i64` is fine. it'd probably be nice to use cursor style instead of offset/limit but we generally use `offset=0` so I think it's pretty much just cursor+limit in practice. 

which makes me question if we actually need these parameters.. could we use the range fenceposts? it seems like we should be able to paginate with either left+limit, or a range where we want everything inside it. I might be forgetting why we need this though.

Reply:
I made a separate PR to address this limit/offset issues https://github.com/ceramicnetwork/rust-ceramic/pull/624

Reply:
Once that merges I'll rebase this PR and follow the same patterns established there.